amr-wind/boundary_conditions/wall_models/LogLaw.H

```diff
@@ -0,0 +1,54 @@
+#ifndef LogLaw_H
+#define LogLaw_H
+
+#include "AMReX_AmrCore.H"
+namespace amr_wind {
+struct LogLaw
+{
+    /*
+     * A simple wall model that sets the wall-shear stress
+     * based on computing u_tau given the horizontal velocity
+     * magnitude at a zref. This is akin to an explicit non-linear
+     * Robin boundary condition at the wall.
+     */
+
+    // Log law constants from Lee & Moser 2015
+    // https://doi.org/10.1017/jfm.2015.268.
+    amrex::Real B{4.27};
+    amrex::Real kappa{0.384};
+    int max_iters = 25; // Max iterations for u_tau Newton-Raphson solve
+    // Reference height for log law
+    amrex::Real zref;
+    amrex::Real nu; // molecular viscosity
+    // u_tau state variable, gets updated in update_utau depending on
+    // the type of wall model used
+    amrex::Real utau_mean{1.0};
+    amrex::Real wspd_mean; // mean horizontal velocity magnitude
+
+    void update_utau_mean() { utau_mean = get_utau(wspd_mean); }
+
+    AMREX_GPU_DEVICE AMREX_FORCE_INLINE amrex::Real
+    get_utau(amrex::Real wspd) const
+    {
+        amrex::Real utau_iter;
+        amrex::Real wspd_pred;
+        amrex::Real wspd_deriv;
+        amrex::Real zplus;
+        amrex::Real utau = utau_mean;
+        int iter = 0;
+        do {
```

Comment:
Haven't seen a do...while in a long time ;) Can you just make this a while loop? Eg. incflo.cpp